File: coding-test/javascript/src/weekly-challenge/minimum-rectangle.js
// @ts-check

/**
 * @param {number[][]} sizes
 * @returns
 */
function solution(sizes) {
  let maxLength = 0
  let minLength = 0

  for (let i = 0; i < sizes.length; i += 1) {
    const max = Math.max(...sizes[i])
    const min = Math.min(...sizes[i])

    if (max > maxLength) {
      maxLength = max
    }

    if (min > minLength) {
      minLength = min
    }
  }

  return maxLength * minLength
}

function diffSolution(sizes) {
  const [hor, ver] = sizes.reduce(
    ([h, v], [a, b]) => [
      Math.max(h, Math.max(a, b)),
      Math.max(v, Math.min(a, b)),
    ],
    [0, 0]
  )
  return hor * ver
}

const sizes = [
  [60, 50],
  [30, 70],
  [60, 30],
  [80, 40],
]
console.log(solution(sizes))
console.log(diffSolution(sizes))

File: practice-coding-test/java-kotlin/src/kr/co/programmers/java/weekly_challenge/MinimumRectangle.java
package kr.co.programmers.java.weekly_challenge;

public class MinimumRectangle {

    public static void main(String[] args) {
        int[][] sizes = {{60, 50}, {30, 70}, {60, 30}, {80, 40}};
        System.out.println(new Solution().solution(sizes));
    }

    static class Solution {
        public int solution(int[][] sizes) {
            int width = 0;
            int height = 0;

            for(int i = 0; i < sizes.length; i++){
                int max = sizes[i][0];
                int min = sizes[i][1];

                if(sizes[i][0] < sizes[i][1]){
                    max = sizes[i][1];
                    min = sizes[i][0];
                }

                if(max > width){
                    width = max;
                }

                if(min > height){
                    height = min;
                }
            }

            return width * height;
        }
    }
}
